src/apiCaller.js

```diff
@@ -1,13 +1,26 @@
 import PageData from './components/pageData'
+import path from 'path'
 
 const apiCaller = (opts) => {
   return (files, metalsmith, done) => {
+    const postQuery = (opts.url && opts.url.postBuild) || false
     new PageData({
       opts,
       files
     }).then((data) => {
-      files = data
-      done()
+      const newDone = data => {
+        files = data
+        done()
+      }
```

Comment:
Do we need this? Can we not do it differently? 

Does `data` change anywhere?

Reply:
I can't see how re-setting `files` prop has an effect inside this function - but the done part should be kept.